### PR TITLE
added example of flashing from python code (ESPTOOL-952)

### PR DESCRIPTION
**Pull request by brentpicasso**
Example app demonstrating how to flash from python code.  Follow up from PR #1022

### CODE REVIEW

**Comment by radimkarnis** [comment]
Hi @brentpicasso,
thanks for taking the time to follow up and contribute to the documentation!

We will be refactoring esptool to be easily usable as a Python module in the next major release, so this script will most likely stop working quite soon due to changing APIs. On the other hand, starting with the documentation now won't hurt; this contribution seems like a good start.

I have some issues and suggestions before we can move on. Feel free to say you don't have time for this, in that case, I can take over.

**Comment by radimkarnis** [comment]
1) The commit name, please see the message from our bot above this comment. The commit type is missing. I suggest something like `docs(scripting): Add example of usage as a Python module".

**Comment by radimkarnis** [comment]
2) New docs page. We already have a page explaining some basic scripting [here](https://docs.espressif.com/projects/esptool/en/latest/esp32/esptool/scripting.html). I would add this as a new section (e.g. `Using Esptool as a Python Module`).

File: docs/en/advanced-topics/flashing-from-python-code.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Flashing from your python code
+==============================
+
+The following is an example on how to flash the ESP32 from a custom application:
```
Comment (radimkarnis) [inline docs/en/advanced-topics/flashing-from-python-code.rst:4]: Changing `ESP32` to `{IDF_TARGET_NAME}` will make it dynamically change based on the selected target.

File: docs/en/advanced-topics/flashing-from-python-code.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# the port of the connected ESP32
+port = "/dev/ttyACM0"
+
+# The firmware file
+filename = "./firmware.bin"
+
+# Typical block size (16 KB)
+BLOCK_SIZE = 0x4000
+
+#beginning of flash address
```
Comment (radimkarnis) [inline docs/en/advanced-topics/flashing-from-python-code.rst:22]: Please keep all the comments formatted uniformly.

File: docs/en/advanced-topics/flashing-from-python-code.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    
+    stub = esp.run_stub()
+    with open(filename, 'rb') as firmware:
+        firmware_data = firmware.read()
+        print(f"firmware length {len(firmware_data)}")
+        total_size = len(firmware_data)
+        stub.flash_begin(total_size, FLASH_BEGIN)
+
+        # Flash in blocks using flash_block
+        block_size = BLOCK_SIZE
```
Comment (radimkarnis) [inline docs/en/advanced-topics/flashing-from-python-code.rst:42]: This is called `FLASH_WRITE_SIZE` in esptool and changes depending on if that's the ROM bootloader or the stub. 
So in this case `block_size = stub.FLASH_WRITE_SIZE` seems better.

File: docs/en/advanced-topics/flashing-from-python-code.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+port = "/dev/ttyACM0"
+
+# The firmware file
+filename = "./firmware.bin"
+
+# Typical block size (16 KB)
+BLOCK_SIZE = 0x4000
+
+#beginning of flash address
+FLASH_BEGIN = 0x10000
```
Comment (radimkarnis) [inline docs/en/advanced-topics/flashing-from-python-code.rst:23]: Suggest renaming to `FLASH_ADDRESS` or `FLASH_OFFSET`

**Review by radimkarnis — CHANGES_REQUESTED** [review]
Left a few suggestions. Let me know if that seems reasonable!

**Comment by brentpicasso** [comment]
Hi, @radimkarnis I ran out of time this week, preparing for some travel.  I accept your offer to move it forward! Thank you!

**Pull request closed without merging**